### DISCUSSION TITLE
Progress bar for `sf apex run test` during synchronous test execution

### DISCUSSION THREAD

**Discussion by dpr-nh** [forcedotcom/cli#2453] [Ideas]
Currently, when running the `sf project deploy start` command like so:
```shell
sf project deploy start -m ApexClass:SomeDummyClass ApexClass:SomeDummyClass_Test --test-level RunSpecifiedTests --tests SomeDummyClass_Test
```
It gives a progress bar while the deployment is in progress:
```
Deploying v56.0 metadata to testuser@testorg.com using the v58.0 SOAP API.
Deploy ID: <deployid>
Status: In Progress | ████████████████████████████████████████ | 2/2 Components (Errors:0) | 0/0 Tests (Errors:0)
```

Would it be feasible to have the same when executing the `sf apex run test` command with the synchronous flag?
```
sf apex run test --synchronous --tests SomeDummyClass_Test
```
During execution it would display something like:
```
Test Run ID: <testrunid>
Status: In Progress | ██████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 3/15 Tests (Errors:0)
```

**Comment by cristiand391** [comment]
Hi @dpr-nh 

`apex run test --synchronous` uses the tooling api to run the tests, specifically the `runTestsSynchronous` endpoint:
https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/intro_rest_resources.htm

`sf project deploy start` gets the progress by polling the deploy in progress, we could add spinner or fake progress bar for `apex run test --synchronous` but it's not possible to how the counter (3/15 tests).
